feat: statistics calculator (#48181)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-advanced-array-methods-by-building-a-statistics-calculator/635090f47eb6d9563a6fed05.md

---
id: 635090f47eb6d9563a6fed05
title: Step 22
challengeType: 0
dashedName: step-22
---

# --description--

The next step is to find the number in the middle of the list. If the list has an odd number of numbers, the middle number is the median. If the list has an even number of numbers, the median is the average of the two middle numbers.

> Declare a `median` variable. Using the ternary operator, check if the length of `array` is even. You can do this by using the <dfn>modulus operator</dfn>, which is represented by the `%` symbol. This operator returns the remainder of the division of two numbers. If the remainder is `0`, the number is even. If the remainder is `1`, the number is odd.
> 
> ```js
> array.length % 2 === 0;
> ```
> 
> If the array is even, find the two middle numbers and calculate the mean of those numbers. If the array is odd, find the middle number and assign it to the `median` variable.

Reviewer: Tried to reorder things so the instructions are at the bottom of the text:

```suggestion
You can check if a number is even or odd with the <dfn>modulus operator</dfn>, which is represented by the `%` symbol. This operator returns the remainder of the division of two numbers. If the remainder is `0`, the number is even. If the remainder is `1`, the number is odd:

```js
array.length % 2 === 0;
```

Declare a `median` variable. Using the ternary operator, check if the length of `array` is even. If the length of `array` is even, find the two middle numbers and calculate the mean of those numbers. If the length of `array` is odd, find the middle number and assign it to the `median` variable.
```
